static/app/views/settings/projectIssueGrouping/upgradeGrouping.tsx

```diff
@@ -67,38 +102,35 @@ function UpgradeGrouping({
     } catch {
       handleXhrErrorResponse(t('Unable to upgrade config'));
     }
-  };
+  }
 
-  if (!groupingConfigs) {
-    return null;
+  function openUpgradeGroupingConfirmModal() {
+    if (location.hash !== `#${upgradeGroupingId}` || noUpdates || !groupingConfigs) {
+      return;
+    }
+
+    openConfirmModal({
+      confirmText,
+      priority,
+      onConfirm: handleConfirmUpgrade,
+      message: modalMessage,
+    });
   }
 
-  function getModalMessage() {
-    return (
-      <Fragment>
-        <TextBlock>
-          <strong>{t('Upgrade Grouping Strategy')}</strong>
-        </TextBlock>
-        <TextBlock>
-          {t(
-            'You can upgrade the grouping strategy to the latest but this is an irreversible operation.'
-          )}
-        </TextBlock>
-        <TextBlock>
-          <strong>{t('New Behavior')}</strong>
-          <div dangerouslySetInnerHTML={{__html: marked(updateNotes)}} />
-        </TextBlock>
-        <TextBlock>
-          <Alert type={alertType}>{riskNote}</Alert>
-        </TextBlock>
-      </Fragment>
-    );
+  function handleOpenUpgradeGroupingConfirmModal() {
+    openConfirmModal({
```

Comment:
If you're going to go this way, why not extract out this object out of the functions

Reply:
Actually, I like the idea more of just having one function that does this named `handleOpenConfirmModal` and the useEffect has the logic that decides if the hash matches.